cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeService.java

```diff
@@ -362,15 +386,16 @@ public Void call() throws Exception {
       }
     }
 
-    String extraClassPath = "$PWD/" + CDAP_SPARK_JAR + "/lib/*";
+    String extraClassPath = "$PWD/" + CDAP_LAUNCHER_JAR + File.pathSeparator + "$PWD/" + CDAP_SPARK_JAR + "/lib/*";
```

Comment:
CDAP_LAUNCHER_JAR is a jar. Thought about using Joiner, but doesn't seems give better code. There are two separators involves. The `pathSeparator == :` and the `separator == /`.